packages/flutter_tools/lib/src/android/native_assets.dart

```diff
@@ -0,0 +1,219 @@
+      target: target,
+      buildMode: buildModeCli,
+      workingDirectory: projectUri,
+      includeParentEnvironment: true,
+      cCompilerConfig: await buildRunner.ndkCCompilerConfig,
+      targetAndroidNdkApi: targetAndroidNdkApi,
+    );
+    ensureNativeAssetsBuildSucceed(result);
+    nativeAssets.addAll(result.assets);
+    dependencies.addAll(result.dependencies);
+  }
+  ensureNoLinkModeStatic(nativeAssets);
+  globals.logger.printTrace('Building native assets for $targets done.');
+  if (isAddToApp && nativeAssets.isNotEmpty) {
+    throwToolExit(
+        'Native assets are not yet supported in Android add2app.');
+  }
+  final Map<Asset, Asset> assetTargetLocations =
+      _assetTargetLocations(nativeAssets);
+  await _copyNativeAssetsAndroid(buildUri_, assetTargetLocations, fileSystem);
+  final Uri nativeAssetsUri = await writeNativeAssetsYaml(
+      assetTargetLocations.values,
+      yamlParentDirectory ?? buildUri_,
+      fileSystem);
+  return (nativeAssetsUri, dependencies.toList());
+}
+
+Future<void> _copyNativeAssetsAndroid(
+  Uri buildUri,
+  Map<Asset, Asset> assetTargetLocations,
+  FileSystem fileSystem,
+) async {
+  if (assetTargetLocations.isNotEmpty) {
+    globals.logger
+        .printTrace('Copying native assets to ${buildUri.toFilePath()}.');
+    for (final String jniArchDir in _architectureStringsCMakeAndroid.values) {
+      final Uri archUri = buildUri.resolve('jniLibs/lib/$jniArchDir/');
+      await fileSystem.directory(archUri).create(recursive: true);
+    }
+    for (final MapEntry<Asset, Asset> assetMapping
+        in assetTargetLocations.entries) {
+      final Uri source = (assetMapping.key.path as AssetAbsolutePath).uri;
+      final Uri target = (assetMapping.value.path as AssetAbsolutePath).uri;
+      final String jniArchDir = _architectureStringsCMakeAndroid[
+          assetMapping.value.target.architecture]!;
+      final Uri archUri = buildUri.resolve('jniLibs/lib/$jniArchDir/');
+      final Uri targetUri = archUri.resolveUri(target);
+      final String targetFullPath = targetUri.toFilePath();
+      await fileSystem.file(source).copy(targetFullPath);
+    }
+    globals.logger.printTrace('Copying native assets done.');
+  }
+}
+
+/// Extract the [Target] from an [AndroidArch].
+Target _getNativeTarget(AndroidArch androidArch) {
+  switch (androidArch) {
+    case AndroidArch.armeabi_v7a:
+      return Target.androidArm;
+    case AndroidArch.arm64_v8a:
+      return Target.androidArm64;
+    case AndroidArch.x86:
+      return Target.androidIA32;
+    case AndroidArch.x86_64:
+      return Target.androidX64;
+  }
+}
+
+Map<Asset, Asset> _assetTargetLocations(List<Asset> nativeAssets) {
+  return <Asset, Asset>{
+    for (final Asset asset in nativeAssets)
+      asset: _targetLocationAndroid(asset),
+  };
+}
+
+Asset _targetLocationAndroid(Asset asset) {
+  final AssetPath path = asset.path;
+  switch (path) {
+    case AssetSystemPath _:
```

Comment:
For my own understanding can you share what is going on with _:. 
All I could find in the dart docs is https://dart.dev/tools/linter-rules/invalid_case_patterns